ui/app/src/services/content.ts

```diff
@@ -807,23 +813,39 @@ export function getChildrenByPath(
   );
 }
 
+export function fetchItemsByPath(siteId: string, paths: string[]): Observable<SandboxItem[]>;
 export function fetchItemsByPath(
-  site: string,
+  siteId: string,
   paths: string[],
-  options?: {
-    skipHomePathOverride?: boolean;
-  }
-): Observable<DetailedItem[]> {
-  const requests: Observable<DetailedItem>[] = [];
-  paths.forEach((path) =>
-    requests.push(
-      getDetailedItem(
-        site,
-        path === '/site/website' ? (options?.skipHomePathOverride ? path : '/site/website/index.xml') : path
-      )
+  options: FetchItemsByPathOptions & { castAsDetailedItem: false }
+): Observable<SandboxItem[]>;
+export function fetchItemsByPath(
+  siteId: string,
+  paths: string[],
+  options: FetchItemsByPathOptions & { castAsDetailedItem: true }
+): Observable<DetailedItem[]>;
+export function fetchItemsByPath(
+  siteId: string,
+  paths: string[],
+  options: FetchItemsByPathOptions
+): Observable<SandboxItem[]>;
+export function fetchItemsByPath(
+  siteId: string,
+  paths: string[],
+  options?: FetchItemsByPathOptions
+): Observable<SandboxItem[] | DetailedItem[]> {
+  const { castAsDetailedItem = false, preferContent = true } = options;
```

Comment:
Need to protect against options being undefined (i.e. can't restructure properties from undefined)
```suggestion
  const { castAsDetailedItem = false, preferContent = true } = options ?? {};
```